go/vt/worker/split_diff.go

```diff
@@ -398,7 +402,7 @@ func (sdw *SplitDiffWorker) diff(ctx context.Context) error {
 		sdw.destinationSchemaDefinition, err = sdw.wr.GetSchema(
 			shortCtx, sdw.destinationAlias, nil /* tables */, sdw.excludeTables, false /* includeViews */)
 		cancel()
-		rec.RecordError(err)
+		sdw.markAsFailed(rec, err)
```

Comment:
I don't think the change in its current form makes sense?

Here and throughout you would replace the RecordError call with RecordError+setting the state to "WorkerStateDiff".

Why do you need to set the state to "WorkerStateDiff" everywhere? That's already done at the top of this method and feels very redundant.

Reply:
Ooops. Well caught...